pkg/danm/danm.go

```diff
@@ -183,13 +191,115 @@ func extractConnections(args *cniArgs) error {
   return nil
 }
 
+func getResourcePrefix(args *cniArgs, resourceType string)(string,error){
+  confArgs, err := loadNetConf(args.stdIn)
+  if err != nil {
+    return "", errors.New("cannot load CNI NetConf due to error:" + err.Error())
+  }
+  k8sClient, err := createK8sClient(confArgs.Kubeconfig)
+  if err != nil {
+    return "", errors.New("cannot create K8s REST client due to error:" + err.Error())
+  }
+  configmap, err := k8sClient.CoreV1().ConfigMaps(string("kube-system")).Get(string("resource-prefix-map"), meta_v1.GetOptions{})
+  if err != nil {
+    return "", errors.New("failed to get 'resource-prefix-map' Config Map data from K8s API server due to:" + err.Error())
+  }
+  return  configmap.Data[resourceType], nil
+}
+
+func getRegisteredDevices(args *cniArgs, cniType string)([]string,error){
+  resourceMap := make(map[string]*checkpoint_types.ResourceInfo)
+  if string(args.podUid) != "" {
+    checkpoint, err := checkpoint_utils.GetCheckpoint()
+    if err != nil {
+      return nil, errors.New("failed to instantiate checkpoint object due to:" + err.Error())
+    }
+    resourceMap, err = checkpoint.GetComputeDeviceMap(string(args.podUid))
+    if err != nil {
+      return nil, errors.New("failed to retrieve Pod info from checkpoint object due to:" + err.Error())
+    }
+  }
+  prefix, err := getResourcePrefix(args, cniType)
```

Comment:
this is how this function should start: if no devices are configured, simply return and go on with network provisioning

Reply:
Code has been refactored